### PR TITLE
A slew of small fixes on the query-execution openAPI spec

### PR DESCRIPTION
**Pull request by msf**
- wording on multiple descriptions was cleaned up.
- adding api_key query parameter
- Improved documentation on the Pagination page

### CODE REVIEW

File: api-reference/overview/pagination.mdx [hunk 1]
```diff
@@ -25,7 +25,7 @@ To paginate through results:
     <Tabs>
         <Tab title="cURL">
         ```bash 
-        curl -X GET 'https://api.dune.com/api/v1/query/3426636/results?limit=5&offset=0' -H 'x-dune-api-key: {{api_key}}' 
+        curl -X GET 'https://api.dune.com/api/v1/query/3426636/results?limit=1000' -H 'x-dune-api-key: {{api_key}}'
```
Comment (msf) [inline api-reference/overview/pagination.mdx:28]: sane defaults on all examples

File: api-reference/overview/pagination.mdx [hunk 2]
```diff
@@ -408,12 +407,17 @@ If you pass in an invalid `limit` and `offset` parameter values, you will get an
...
 
-Dune imposes a 1GB data limit on query results for each query execution. If your query yields more than 1GB of data, the result will be truncated in storage. In such cases, pulling the result data with `allow_partial_results` set to false (the default) will trigger an error message: "error": "Partial Result, please request with 'allows_partial_results=true'". If you wish to retrieve partial results, you can pass the parameter `allow_partial_results=true`. But please make sure you indeed want to fetch the truncated result.
+
+**Data Size Limit **
+
+Dune internally has a maximum query result size limit (which currently is 8GB, but subject to increase in the future). If your query yields more than 8GB of data, the result will be truncated in storage. In such cases, pulling the result data (using pagination) but without specifying `allow_partial_results` set to true will trigger an error message: "error": "Partial Result, please request with 'allows_partial_results=true'". If you wish to retrieve partial results, you can pass the parameter `allow_partial_results=true`. But please make sure you indeed want to fetch the truncated result.
```
Comment (msf) [inline api-reference/overview/pagination.mdx:416]: @agaperste , @andrewhong5297 I added this longer explanation of the query result size limit, and this part over here.

I see in other pages references to 1GB, where we have that we should mention 8GB and include a link to this section of this page.

Comment (agaperste) [reply api-reference/overview/pagination.mdx:416]: Yep when we wrote this there was a bug with result size limit increasing to 8GB so kept it 1GB. I can update other references

Also, wow, didn't know server will automatically override 500k rows to 3000 rows.. is that dynamic? or a hard limit on how many rows user should specify?

also side note, the formatting is slightly off, let me just fix it alongside with fixing the 1GB limit in a min
image

Comment (msf) [reply api-reference/overview/pagination.mdx:417]: > Also, wow, didn't know server will automatically override 500k rows to 3000 rows.. is that dynamic? or a hard limit on how many rows user should specify?

The problem is:
- there must be an upper limit, or some client might request limit=8_000_000

There's two options:
- reject the request and tell the user to use a lower value (annoying!)
- accept, but change the limit to acceptable value (if we know the solution,  we just apply it for the user and avoid annoying him)

Now the follow up is:
- what is the acceptable value for the limit? 

We know we want the API to be friendly to mobile and phones, so we don't want a max value that will take a long time to load on 3G or 4G nor blow up the mobile browser memory usage. The exact max value will require exploration, setting it in stone in the documentation might lead to wrong assumptions. The API provides all the information required for the APP (or the developer) to use the API correctly and smoothly without knowing the exact max value.

**Pull request merged**